src/snowflake/snowpark/dataframe.py

```diff
@@ -610,21 +685,44 @@ def collect(
         See also:
             :meth:`collect_nowait()`
         """
+
+        kwargs = {}
+        if _emit_ast:
+            # Add an Assign node that applies SpDataframeCollect() to the input, followed by its Eval.
+            repr = self._session._ast_batch.assign()
+            expr = with_src_position(repr.expr.sp_dataframe_collect)
+            debug_check_missing_ast(self._ast_id, self)
+            expr.id.bitfield1 = self._ast_id
+            if statement_params is not None:
+                build_expr_from_dict_str_str(expr.statement_params, statement_params)
+            expr.block = block
+            expr.case_sensitive = case_sensitive
+            expr.log_on_exception = log_on_exception
+            expr.no_wait = False
+
+            self._session._ast_batch.eval(repr)
+
+            # Flush the AST and encode it as part of the query.
+            _, kwargs["_dataframe_ast"] = self._session._ast_batch.flush()
```

Comment:
Can we create a constant for `"_dataframe_ast"`?

Reply:
add `DATAFRAME_AST_PARAMETER="_dataframe_ast"`